libraries/chef_vault_secret.rb

```diff
@@ -45,9 +45,8 @@ class ChefVaultSecret < Chef::Resource::LWRPBase
 
   module Provider
     class ChefVaultSecret < Chef::Provider::LWRPBase
-      provides(:chef_vault_secret)
-
-      use_inline_resources if defined?(:use_inline_resources)
+      provides(:chef_vault_secret) if defined?(provides)
+      use_inline_resources if defined?(use_inline_resources)
```

Comment:
defensive coding around use_inline_resources is a chef-10-ism that we've definitely ripped out of cookbooks and stopped doing since chef-10 is dead.